src/app/pagination/pagination.component.ts

```diff
@@ -0,0 +1,222 @@
+  DoCheck,
+  OnChanges,
+  SimpleChanges,
+  Input,
+  Output,
+  EventEmitter,
+  ViewEncapsulation,
+  OnInit
+} from '@angular/core';
+
+import { cloneDeep, defaults, isEqual } from 'lodash';
+
+import { PaginationConfig } from './pagination-config';
+
+/**
+ * Component for rendering pagination
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-pagination',
+  templateUrl: './pagination.component.html'
+})
+
+export class PaginationComponent implements OnInit, DoCheck, OnChanges {
+
+  /**
+   * The Pagination config contaning component properties
+   */
+  @Input() config: PaginationConfig;
+
+  /**
+   * The Event is emitted when Page Size is changed
+   */
+  @Output('onPageSizeChange') onPageSizeChange = new EventEmitter();
+
+  /**
+   * The Event is emitted when Page Number is Changed
+   */
+  @Output('onPageNumberChange') onPageNumberChange = new EventEmitter();
+
+
+  private defaultConfig = {
+    pageNumber: 1,
+    pageSizeIncrements: [5, 10, 20, 40, 80, 100],
+    pageSize: 5
+  } as PaginationConfig
+  private prevConfig: PaginationConfig;
+  private _lastPageNumber: number;
+
+  /**
+   * The default constructor
+   */
+  constructor() { }
+
+  // Initialization
+
+  /**
+   *  Setup component configuration upon initialization
+   */
+  ngOnInit(): void {
+    this.setupConfig();
+    this.lastPageNumber = this.getLastPageNumber();
+  }
+
+  /**
+   *  Check if the component config has changed
+   */
+  ngDoCheck(): void {
+    // Do a deep compare on config
+    if (!isEqual(this.config, this.prevConfig)) {
+      this.setupConfig();
+    }
+  }
+
+  /**
+   * Called when properties value are changed
+   * @param changes obj of type SimpleChanges containing old and new value
+   */
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.totalItems && !changes.totalItems.isFirstChange()) {
+      this.lastPageNumber = this.getLastPageNumber();
+      this.gotoFirstPage();
+    }
+  }
+
+  /**
+   * Setup default config
+   */
+  protected setupConfig(): void {
+    if (this.config !== undefined) {
+      defaults(this.config, this.defaultConfig);
+    } else {
+      this.config = cloneDeep(this.defaultConfig);
+    }
+    this.prevConfig = cloneDeep(this.config);
+  }
+
+  /**
+   * get _lastPageNumber
+   */
+  get lastPageNumber(): number {
+    return this._lastPageNumber;
+  }
+
+  /**
+   * set _lastPageNumber
+   */
+  set lastPageNumber(value: number) {
+    this._lastPageNumber = value;
+  }
+
+  /**
+   * Page size is changed
+   * @param newPageSize new page size
+   */
+  private onPageSizeUpdate($event: Event, newPageSize: number): void {
+    this.config.pageSize = newPageSize;
+    this.lastPageNumber = this.getLastPageNumber();
+    this.gotoFirstPage();
+    this.onPageSizeChange.emit({
+      pageSize: newPageSize
+    });
+  }
+
+  /**
+   * Page number is changed using input field
+   */
+  public onPageNumberUpdate($event: KeyboardEvent, value: string): void {
```

Comment:
Per my previous review, you still need to check the event to ensure the user pressed the enter key.

Please see my earlier comments -- I believe that I provided an example?